src/acl/Url.cc

```diff
@@ -11,16 +11,15 @@
 #include "squid.h"
 #include "acl/FilledChecklist.h"
 #include "acl/Url.h"
+#include "anyp/Uri.h"
```

Comment:
> Amos: Two things for this specific case:
> 
> 1. IMO, the conversion to `Decode()` should be done only where the `rfc1738_unescape()` is forcing use of `char*` in a series of `SBuf` operations.

I disagree: IMO, **all** rfc1738_unescape() should be eventually converted (or removed, etc.). We may prioritize certain use cases over others, of course, but the overall goal should be to get rid of the very problematic rfc1738_unescape().

> This case we have a necessary conversion to `char*` regardless of the decode.

SBuf returned by Decode() can be converted to a c-string so the necessity of that conversion alone is not a disqualifying condition.

> So not (yet) a good candidate for removal of `rfc1738_unescape()`.

That may still be true (I do not know), but not for the reasons given.

> 2. The `effectiveRequestUri()` in current v7 code is normalized with inappropriate encoding removed.

What do you mean by "inappropriate" here?

I do not see any decoding in effectiveRequestUri(). In fact, the method _encodes_ userinfo (via Uri::absolute()) and probably uses _encoded_ URL whitespace in URI_WHITESPACE_ENCODE cases (see AnyP::Uri::parse()). What am I missing?

Reply:
> I disagree: IMO, **all** rfc1738_unescape() should be eventually converted (or removed, etc.). We may prioritize certain use cases over others, of course, but the overall goal should be to get rid of the very problematic rfc1738_unescape().

This is my general intent

Reply:
> What I am getting from all this is that our goal contexts still differ despite the weekly meetings discussion of what the goal and approach should be.

I do not recall any discussions that contradicted what I have said in my response. I also do not recall us finalizing "goal and approach" of rfc1738_unescape() replacements in any relevant way. If you believe that we have agreed on something specific that (is relevant to this or related future PRs and) contradicts my comments, then I recommend posting that alleged agreement for a confirmation/review. In my experience, the probability of the three of us having the _same_ interpretation of an informal verbal discussion about complex technical matters is asymptotically approaching zero.

> Also, refer to meaning of "IMO".

AFAICT, the meaning of the phrase "in my opinion" does not contradict what I have said. I do not know why "IMO meaning" is mentioned twice in your response. If you feel it is important to clarify this, please detail your concerns.